From e332ca20095fccfea9478650120c37f03fb500fd Mon Sep 17 00:00:00 2001
From: Alasdair Campbell <alcoheca@gmail.com>
Date: Wed, 6 Jun 2012 18:02:16 +0100
Subject: [PATCH 03/21] [UPnP] reinstated: 'fixed: upnp server's http server
 would not provide content length on HEAD requests' by
 elupus @ 9cbcf5063c3af350d385f1992017f6883f37a8e0

---
 lib/libUPnP/Platinum/Source/Core/PltHttpServerTask.cpp | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/lib/libUPnP/Platinum/Source/Core/PltHttpServerTask.cpp b/lib/libUPnP/Platinum/Source/Core/PltHttpServerTask.cpp
index 11a7ac0..7ffed7e 100644
--- a/lib/libUPnP/Platinum/Source/Core/PltHttpServerTask.cpp
+++ b/lib/libUPnP/Platinum/Source/Core/PltHttpServerTask.cpp
@@ -279,7 +279,7 @@ PLT_HttpServerSocketTask::Write(NPT_HttpResponse* response,
     // get the request entity to set additional headers
     NPT_InputStreamReference body_stream;
     NPT_HttpEntity* entity = response->GetEntity();
-    if (entity && NPT_SUCCEEDED(entity->GetInputStream(body_stream))) {
+    if (entity) {
         // set the content length if known
         if (entity->ContentLengthIsKnown()) {
             headers.SetHeader(NPT_HTTP_HEADER_CONTENT_LENGTH, 
@@ -341,7 +341,7 @@ PLT_HttpServerSocketTask::Write(NPT_HttpResponse* response,
     NPT_CHECK_WARNING(output_stream->WriteFully(header_stream.GetData(), header_stream.GetDataSize()));
 
     // send response body if any
-    if (!headers_only && !body_stream.IsNull()) {
+    if (!headers_only && NPT_SUCCEEDED(entity->GetInputStream(body_stream)) && !body_stream.IsNull()) {
         NPT_CHECK_WARNING(NPT_StreamToStreamCopy(
             *body_stream.AsPointer(), 
             *output_stream.AsPointer(),
-- 
1.7.11.msysgit.0

